src/rpc/rawtransaction.cpp

```diff
@@ -554,7 +554,14 @@ UniValue decodescript(const UniValue& params, bool fHelp)
     }
     ScriptPubKeyToJSON(script, r, false);
 
-    r.push_back(Pair("p2sh", CBitcoinAddress(CScriptID(script)).ToString()));
+    UniValue type;
+    type = find_value(r, "type");
+
+    if ((!type.isNull()) && (type.isStr()) && (type.get_str() != "scripthash"))
```

Comment:
Should probably drop the isNull check because a value can't be both a string and a null at the same time.

Also maybe drop the parens around && conditions (searching for "&&" in this file shows no parens used in the other cases).